### PR TITLE
Add artifactory resolver

### PR DESCRIPTION
**Pull request by felixmulder**
This change adds the local EPFL Artifactory for CI integration tests. Which should reduce our daily bandwidth consumption from 50gigs ot less :tada:

### CODE REVIEW

**Comment by smarter** [comment]
> Which should reduce our daily bandwidth consumption from 50gigs ot less :tada:

Isn't that just because we haven't updated the ivy2 cache on the CI in a while?

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,6 +2,12 @@
 //
 // e.g. addSbtPlugin("com.github.mpeltonen" % "sbt-idea" % "1.1.0")
 
+// Add EPFL Artifactory if on CI
+resolvers ++= {
```
Comment (smarter) [inline project/plugins.sbt]: I think you're adding the same resolver 5 times, which may impact resolving speed.

**Comment by felixmulder** [comment]
That's also true - but using this, even if we get out of sync - it should still be fine.

Regarding adding the resolver in multiple places - it seems to be needed for resolving plugins... T_T

File: project/plugins.sbt (hunk 1 above)
Comment (smarter) [inline project/plugins.sbt]: Also you're not adding it to all the scripted tests (sbt-dotty/sbt-test/...), is that needed to?

**Comment by smarter** [comment]
Yeah, but keeping in sync is not a lot of work, compared to maintaining a whole artifactory instance, I don't get the rationale here.

File: project/plugins.sbt (hunk 1 above)
Comment (felixmulder) [inline project/plugins.sbt]: It might be. Maybe we should just include it in `Build.scala` and then update our cache. WDYT?

**Comment by felixmulder** [comment]
What is the maintenance cost?

**Comment by smarter** [comment]
What's the plan if the artifactory instance breaks? Do we all need to learn how it works? Can we disable it in some way if we need to? etc.

**Comment by felixmulder** [comment]
If it's down, then it's down - it'll fall back to the other resolvers.

Fabien is running it, disabling == remove from script

**Comment by smarter** [comment]
It looks like artifactory is not even open source (I looked around and couldn't find the sources at least) that's a deal breaker for me. I also don't understand why we would need to maintain a giant Java application when all we need is a 10 lines bash script to periodically check if we need to update the ivy2 cache and regenerated the docker image.

**Comment by felixmulder** [comment]
The difference is that somebody still needs to do that - if you want to, that'd be great - but this fix is now.

There is also the issue of drone being tied to specific tags of the docker images it uses as its testing environment. So we'd need the script to also update `.drone.yml`, open a PR and have someone merge it (or it can commit straight to master 🤘 😈 )

We could also let the script push to the `latest` tag as well as a date tag, then we'd have the `.drone.yml` file point to latest and bisect if there's something wrong with the cache.

I'm fine with either one here really, the effect from my vantage point is the same...

**Comment by DarkDimius** [comment]
An open source solution could be to set up a reverse caching proxy(e.g. squid) on EPFL infrastructure.

I'm fine with artifactory though. Sonatype & Maven central are not open-source either. They have free-ish versions, but the thing that actually runs is highly modified.

**Comment by smarter** [comment]
Since sbt does not download anything in parallel, using artifactory will also be slower than having an up-to-date local cache, in adition to being much more complicated.

**Comment by OlivierBlanvillain** [comment]
Counter proposal: run the tests offline to make sure that the cache is fully up to date. That would also give ultimate reproducibility of history and immunity to external failures.

**Pull request closed without merging**